homeassistant/components/bluetooth/active_update_processor.py

```diff
@@ -7,26 +7,28 @@
 
 from collections.abc import Callable, Coroutine
 import logging
-from typing import Any, Generic, TypeVar
+from typing import TYPE_CHECKING, Any
 
 from bleak import BleakError
 from bluetooth_data_tools import monotonic_time_coarse
+from typing_extensions import TypeVar
 
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.debounce import Debouncer
 
 from . import BluetoothChange, BluetoothScanningMode, BluetoothServiceInfoBleak
 from .passive_update_processor import PassiveBluetoothProcessorCoordinator
 
+if TYPE_CHECKING:
+    from sensor_state_data import SensorUpdate  # noqa: F401
```

Comment:
Was also thinking about this. In practice though, only `aranet` uses something different with `Aranet4Advertisement`. If the dependency would be typed, mypy would have emitted an error in case the default was used. That's why I decided to go with it.

Reply:
`sensor_state_data` also isn't a dep of `bluetooth` so it seems out of place to use it here.

Reply:
> `sensor_state_data` also isn't a dep of `bluetooth` so it seems out of place to use it here.

That's why I moved it to the `if TYPE_CHECKING` block. We already do something similar in `helpers/sensor.py`.
https://github.com/home-assistant/core/blob/9f321642b2c0ac8dc772da58a2e23b83897d3541/homeassistant/helpers/sensor.py#L11-L15

Reply:
Removed the `TypeVar` defaults to unblock this PR.